controllers/datafoundationclient.go

```diff
@@ -47,16 +50,29 @@ func dfcSetupWatches(controllerBuilder *builder.Builder) {
 		),
 	)
 	controllerBuilder.
-		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
-		Owns(&ocsclient.StorageClient{}).
-		Owns(&ocsclient.StorageClassClaim{})
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates)
+
+	if utils.MapCRDAvailability(storageClientCRD)[storageClientCRD] {
+		doesCRDExistsAtStart = true
+		controllerBuilder.
+			Owns(&ocsclient.StorageClient{}).
+			Owns(&ocsclient.StorageClassClaim{})
+	}
 }
 
 func DFCAddToScheme(scheme *runtime.Scheme) {
 	utilruntime.Must(ocsclient.AddToScheme(scheme))
 }
 
 func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	// Check if the required CRDs are installed or not
+	if !doesCRDExistsAtStart {
+		if utils.MapCRDAvailability(storageClientCRD)[storageClientCRD] {
+			offeringReconciler.Log.Info("operator needs to restart to register watchers for offering CRDs")
+			os.Exit(1)
```

Comment:
Don't exist from within an offering code. Return a predefined error that the outer reconcile code could translate into  an `os.Exist`